client/src/lib.rs

```diff
@@ -285,13 +277,16 @@ impl Client {
 			let denominator = score_rat.denom().to_string();
 			let score = score_rat.to_integer().to_string();
 
-			formatted_scores.push(vec![
-				peer_address, score_fr_hex, numerator, denominator, score,
-			]);
+			let record =
+				ScoreRecord::new(peer_address, score_fr_hex, numerator, denominator, score);
+
+			formatted_scores.push(record);
 		}
 
-		// Write scores to a CSV file
-		create_csv_file("scores", &formatted_scores).unwrap();
+		// Save
+		CSVFileStorage::<ScoreRecord>::new("../data/scores.csv".into())
+			.save(formatted_scores)
+			.unwrap();
```

Comment:
The library should not hardcode specific implementation, instead we should pass a generic type parameter and call save on that function.
Also it should not know about the location of the file.